### PR TITLE
Introduce smooth quant method and enable smooth quant in language modeling examples

### PR DESCRIPTION
**Pull request by PenghuiCheng**
# What does this PR do?
Introduce smooth quant method in optimization_inc.mdx and enable smooth quant in language modeling examples.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by hshen14** [comment]
@echarlaix Please review and let us know whether it's ok to be merged.

File: examples/neural_compressor/language-modeling/README.md [hunk 1]
```diff
@@ -53,6 +56,25 @@ python run_clm.py \
...
+    --smooth_quant \
+    --apply_pruning \
+    --target_sparsity 0.02 \
+    --num_train_epochs 4 \
+    --max_train_samples 100 \
+    --do_train \
+    --do_eval \
+    --verify_loading \
+    --output_dir /tmp/clm_output
+```
```
Comment (echarlaix) [inline examples/neural_compressor/language-modeling/README.md:76]: Suggesting to show a post-training example instead: 
```suggestion
The following example shows how to apply post-training static quantization using the SmoothQuant methodology on a GPT-Neo model :
```bash
python run_clm.py \
    --model_name_or_path EleutherAI/gpt-neo-125M \
    --dataset_name wikitext \
    --dataset_config_name wikitext-2-raw-v1 \
    --apply_quantization \
    --quantization_approach static \
    --smooth_quant \
    --do_eval \
    --verify_loading \
    --output_dir /tmp/clm_output
```
```

File: examples/neural_compressor/language-modeling/README.md [hunk 2]
```diff
@@ -21,6 +21,9 @@ and [`run_mlm.py`](https://github.com/huggingface/optimum-intel/blob/main/exampl
 allow us to apply different quantization approaches (such as dynamic, static and aware-training quantization) as well as pruning 
 using the [Intel Neural Compressor ](https://github.com/intel/neural-compressor) library for language modeling tasks.
 
+For post training quantization, we can apply recipe of smooth_quant to quantization.
+Usually, smooth quant improves the accuracy of the model.
+
```
Comment (echarlaix) [inline examples/neural_compressor/language-modeling/README.md:26]: The SmoothQuant methodology is also available for post-training quantization.

File: docs/source/optimization_inc.mdx [hunk 3]
```diff
@@ -106,6 +106,33 @@ quantizer.quantize(
     calibration_dataset=calibration_dataset,
     save_directory=save_dir,
 )
+
+### Specify Quantization Recipes
```
Comment (echarlaix) [inline docs/source/optimization_inc.mdx:110]: I would prefer to add it in the documentation once the integration is validated on our side

**Review by echarlaix — COMMENTED** [review]
Thanks a lot for the addition @PenghuiCheng !

File: docs/source/optimization_inc.mdx [hunk 4]
```diff
@@ -106,7 +106,36 @@ quantizer.quantize(
...
+| smooth_quant_args | ✅ |
+
+The SmoothQuant methodology is available for post-training quantization.
+[SmoothQuant](https://arxiv.org/abs/2211.10438), a training free post-training quantization (PTQ) solution,
+offline migrates this difficulty from activations to weights with a mathematically equivalent transformation.
+Usually, smooth quant improves the accuracy of the model.
+
+Please refer to [smooth quant usage](https://github.com/intel/neural-compressor/blob/master/docs/source/smooth_quant.md) and [neural_compressor recipes](https://github.com/intel/neural-compressor/blob/master/docs/source/quantization.md#specify-quantization-recipes).
+
+Example of recipe([link](../../examples/neural_compressor/language-modeling/README.md)):
```
Comment (echarlaix) [inline docs/source/optimization_inc.mdx:125]:
```suggestion

The [SmoothQuant](https://arxiv.org/abs/2211.10438) methodology is available for post-training quantization. This methodology usually improves the accuracy of the model in comparison to other post-training static quantization methodologies. This is done by migratesing the difficulty from activations to weights with a mathematically equivalent transformation.
```

File: docs/source/optimization_inc.mdx [hunk 5]
```diff
@@ -106,7 +106,36 @@ quantizer.quantize(
...
+    "smooth_quant": True,
+    "smooth_quant_args": {
+        "alpha": 0.5,  # "auto" or a float value. Default is 0.5
+        "folding": True,  # folding=False: Allow inserting mul to update the input distribution and not absorbing. It is only for IPEX backend.
+                          # folding=True: Only allow inserting mul with the input scale that can be absorbed into the last layer. 
+                          # If folding not set in config, the default value is IPEX: False (True if version<2.1), Stock PyTorch: True.
+    },
+}
+conf = PostTrainingQuantConfig(recipes=recipes)
 ```
```
Comment (echarlaix) [inline docs/source/optimization_inc.mdx:137]:
```suggestion
```diff
- quantization_config = PostTrainingQuantConfig(approach="static")
+ recipes={"smooth_quant": True,  "smooth_quant_args": {"alpha": 0.5, "folding": True}}
+ quantization_config = PostTrainingQuantConfig(approach="static", backend="ipex", recipes=recipes)
```
```

File: docs/source/optimization_inc.mdx [hunk 6]
```diff
@@ -106,7 +106,36 @@ quantizer.quantize(
...
+Validated Models with smooth quant: [models](https://github.com/intel/neural-compressor/blob/master/docs/source/smooth_quant.md#validated-models)
```
Comment (echarlaix) [inline docs/source/optimization_inc.mdx:138]:
```suggestion
Please refer to INC [documentation](https://github.com/intel/neural-compressor/blob/master/docs/source/smooth_quant.md) and the list of [models](https://github.com/intel/neural-compressor/blob/master/docs/source/smooth_quant.md#validated-models) quantized with the methodology for more details. 
```

File: tests/neural_compressor/test_optimization.py [hunk 7]
```diff
@@ -219,6 +219,46 @@ def preprocess_function(examples, tokenizer):
 
             self.assertTrue(torch.equal(model_outputs["logits"], transformers_outputs["logits"]))
 
+    def test_ipex_static_quantization_with_smoothquant(self):
```
Comment (echarlaix) [inline tests/neural_compressor/test_optimization.py:222]: The only difference with the `test_ipex_static_quantization` is the configuration, could we merge two in one ?

Comment (PenghuiCheng) [reply tests/neural_compressor/test_optimization.py:184]: Yes, I only kept one test case now. thanks!

File: docs/source/optimization_inc.mdx (hunk 3 above)
Comment (PenghuiCheng) [reply docs/source/optimization_inc.mdx:111]: will merge this first, and Let's discuss how to improve later

**Comment by hshen14** [comment]
@mfuntowicz  it seems the issue is not related with INC. Shall we go and merge it?

**Comment by mfuntowicz** [comment]
LGTM! 

I rebased the PR to validate the CI is green after we merged the transformers < 4.29 constraint.

Thanks!
